site/docs/v1/tech/tutorials/integrate-flutter.md

```diff
@@ -0,0 +1,747 @@
+    if (storedRefreshToken == null) return;
+
+    setState(() {
+      isBusy = true;
+    });
+
+    try {
+      final TokenResponse response = await appAuth.token(TokenRequest(
+        FUSION_AUTH_CLIENT_ID,
+        FUSION_AUTH_REDIRECT_URI,
+        issuer: FUSION_AUTH_ISSUER,
+        refreshToken: storedRefreshToken,
+      ));
+
+      final Map<String, Object> profile =
+          await getUserDetails(response.accessToken);
+
+      await secureStorage.write(
+          key: 'refresh_token', value: response.refreshToken);
+      var gravatar = Gravatar(profile['email']);
+      var url = gravatar.imageUrl(
+        size: 100,
+        defaultImage: GravatarImage.retro,
+        rating: GravatarRating.pg,
+        fileExtension: true,
+      );
+      setState(() {
+        isBusy = false;
+        isLoggedIn = true;
+        name = profile['given_name'];
+        picture = url;
+      });
+    } on Exception catch (e, s) {
+      debugPrint('error on refresh token: $e - stack: $s');
+      await logoutAction();
+    }
+  }
+
+  Future<void> logoutAction() async {
+    await secureStorage.delete(key: 'refresh_token');
+    setState(() {
+      isLoggedIn = false;
+      isBusy = false;
+    });
+  }
+}
+
+class Login extends StatelessWidget {
+  final Future<void> Function() loginAction;
+  final String loginError;
+
+  const Login(this.loginAction, this.loginError, {Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: <Widget>[
+        ElevatedButton(
+          onPressed: () async {
+            await loginAction();
+          },
+          child: const Text('Login'),
+        ),
+        Text(loginError ?? ''),
+      ],
+    );
+  }
+}
+
+class Profile extends StatelessWidget {
+  final Future<void> Function() logoutAction;
+  final String name;
+  final String picture;
+
+  const Profile(this.logoutAction, this.name, this.picture, {Key key})
+      : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: <Widget>[
+        Container(
+          width: 200,
+          height: 200,
+          decoration: BoxDecoration(
+            border: Border.all(color: Colors.orange, width: 4),
+            shape: BoxShape.circle,
+            image: DecorationImage(
+              fit: BoxFit.fill,
+              image: NetworkImage(picture ?? ''),
+            ),
+          ),
+        ),
+        const SizedBox(height: 24),
+        Text('Name: $name'),
+        const SizedBox(height: 48),
+        ElevatedButton(
+          onPressed: () async {
+            await logoutAction();
+          },
+          child: const Text('Logout'),
+        ),
+      ],
+    );
+  }
+}
+
+```
+
+That's a lot. Let's break it down section by section. First, inside the `main.dart` file, you need to import all necessary packages:
```

Comment:
We don't need this for the quickstart. You can remove this 'section by section' section.